### ISSUE TITLE
Lines are not drawn correctly during IME edit line operations

### ISSUE THREAD

**Issue by miniksa** [microsoft/terminal#803]
Ported from MSFT: 20973297

> The underscores and the cursor bar aren't being transmitted or drawn correctly when working with the IME. Compare to v1 behavior.
> 
> I think there's something wrong with the communication of both the underscores and the positioning of the caret after the refactoring to use the unified output buffer. Investigate.

**Comment by j4james** [comment]
Btw, I was also looking at this at the same time as issue #810, and I managed to get something working as a proof-of-concept, but the solution I had was horrible, and I'm inclined to think that this area of code might need some rearchitecting which is above my pay grade. But if you're desperate for a better-than-nothing solution (assuming you don't already have a plan in mind), I could probably put together a PR with what I've got.

**Comment by miniksa** [comment]
@j4james... Oh, this code is horrible and you're right that it probably needs a rearchitecting, but given that it's the IME in conhost which is an interactive component.... and our roadmap for interactivity is to move it all to Terminal... we're not really likely to rearchitect anything on that side of the fence.

That said, I looked at it this morning and I came up with a solution to the problem. It's hacky too, might even be the same as your proposed idea, but it's probably best if I bless the solution given I'm the one who fubar'd it all up in the first place. I'll send the PR shortly. Let me know how close I was to your idea.
